stable/ark/templates/deployment.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "ark.fullname" . }}
+  release: {{ .Release.Name }}
+  app: {{ template "ark.name" . }}
+  chart: {{ template "ark.chart" . }}
+  heritage: {{ .Release.Service }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        component: ark
+        release: {{ .Release.Name }}
+        app: {{ template "ark.name" . }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+    {{- end }}
+    spec:
+      restartPolicy: Always
+      serviceAccountName: {{ template "ark.serverServiceAccount" . }}
+      containers:
+        - name: ark
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: Always
+          command:
+            - /ark
+          args:
+            - server
+          {{- if eq .Values.configuration.cloudprovider "azure" }}
+          envFrom:
+            - secretRef:
+                name: {{ .Values.secret.name }}
+          {{- end }}
+          volumeMounts:
+            - name: plugins
+              mountPath: /plugins
+          {{- if (or (eq .Values.configuration.cloudprovider "aws") (eq .Values.configuration.cloudprovider "gcp")) }}
+            - name: cloud-credentials
+              mountPath: /credentials
+          env:
+            - name: {{ .Values.configuration.credentials }}
+              value: /credentials/cloud
+          {{- end }}
+      volumes:
+        {{- if (or (eq .Values.configuration.cloudprovider "aws") (eq .Values.configuration.cloudprovider "gcp")) }}
+        - name: cloud-credentials
+          secret:
+            secretName: {{ .Values.secret.name }}
+        {{- end }}
+        - name: plugins
+          emptyDir: {}
+      {{- if eq .Values.configuration.cloudprovider "azure" }}
+      nodeSelector:
+        beta.kubernetes.io/os: linux
+{{ toYaml .Values.nodeSelector | indent 8 }}
```

Comment:
Node selector should not depend on Azure. 

And as pointed out earlier, I'd remove the special selector for Azure. It is only necessary when you run on Azure and have Windows node. In this case, we can expect the user to configure that.